### PR TITLE
fix golints G114 and package-comments

### PR DESCRIPTION
**Pull request by shalier**
**Description**:
changes:
1. Suppress for demo apps G114: Use of net/http serve function that has no support for setting timeouts
2. Ignore package-comments, known false positive https://golangci-lint.run/usage/false-positives/#exc0015
3. Change golangci-lint timeout to 20m
4. Remove pkg/mesh and moves isValidUUID() to pkg/cli/proxy_get.go
5. Remove unused pkg/k8s/fake folder

Signed-off-by: Shalier Xia <shalierxia@microsoft.com>

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| Other                      | [x ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

6. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?

### CODE REVIEW

File: demo/cmd/bookbuyer/bookbuyer.go [hunk 1]
```diff
@@ -76,8 +76,16 @@ func debugServer() {
 		router.HandleFunc(h.path, h.fn).Methods(h.method)
 	}
 	http.HandleFunc("/favicon.ico", func(w http.ResponseWriter, r *http.Request) {})
+
+	// Initialize bookbuyer HTTP server
+	server := &http.Server{
+		Addr:              fmt.Sprintf(":%d", *port),
+		Handler:           router,
+		ReadHeaderTimeout: time.Second * 10,
+	}
```
Comment (shashankram) [inline demo/cmd/bookbuyer/bookbuyer.go:85]: Is this being done just to please the linter warning? I don't see anything wrong with the previous code as it uses the default mux server from the http pkg. If that's the case, can't we simply suppress the lint warning?

**Comment by jsturtevant** [comment]
I think this linter issue occurs because we are running latest linter in CI. There is lots of value to that, but the UI is pretty rough since it doesn't show you all the linter failures in one go and it takes a really long time locally.  I fixed a bunch of these last month https://github.com/openservicemesh/osm/pull/4969.  

We have the timeout set to 10m in the  https://github.com/openservicemesh/osm/blob/157a15ef7ebc99d48f60f93ec5a4a31a11a15f4e/.golangci.yml#L3 but looks like it only running for 5.  maybe add `args: --timeout=10m` to https://github.com/openservicemesh/osm/blob/157a15ef7ebc99d48f60f93ec5a4a31a11a15f4e/.github/workflows/main.yml#L55-L58 and it will not timeout?

Thanks for addressing them @shalier, this was frustrating when I did it last month :-/

**Comment by jsturtevant** [comment]
That did make it run for 10 mins but looks like it still timed out :-(

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -56,6 +56,7 @@ jobs:
         with:
           version: latest
           skip-pkg-cache: true
+          args: --timeout=20m
```
Comment (keithmattix) [inline .github/workflows/main.yml:59]: Is this required given the .golangci file?

File: pkg/cli/proxy_get.go [hunk 3]
```diff
@@ -72,7 +73,7 @@ func getPortForwarder(clientSet kubernetes.Interface, config *rest.Config, names
 	if err != nil {
 		return nil, fmt.Errorf("error getting pod %s/%s", namespace, podName)
 	}
-	if !mesh.ProxyLabelExists(*pod) {
```
Comment (keithmattix) [inline pkg/cli/proxy_get.go:75]: This looks like a functional change; I'd prefer for functional changes to be in a separate PR from linter appeasement

**Review by keithmattix — APPROVED** [review]

**Review by shashankram — APPROVED** [review]
